app/models/document.rb

```diff
@@ -104,12 +104,23 @@ def humanized_attributes
     end
   end
 
+  def self.extract_body_from_payload(payload)
+    body_attribute = payload['details']['body']
+
+    case
+    when body_attribute.class == Array
+      body_attribute[0]['content']
```

Comment:
We also need to take into account Matilda's comment, ie we don't know for sure that the govspeak is always the first element.